app/src/common/shared/com/igalia/wolvic/ui/widgets/WindowWidget.java

```diff
@@ -1701,7 +1701,12 @@ public void onExternalResponse(@NonNull WSession aSession, @NonNull WWebResponse
         if (!UrlUtils.isFileUri(webResponseInfo.uri())) {
             DownloadJob job;
             if (UrlUtils.isBlobUri(webResponseInfo.uri())) {
-                job = DownloadJob.fromUri(webResponseInfo.uri(), webResponseInfo.headers(), webResponseInfo.body());
+                String uri = webResponseInfo.uri();
+                if (uri.startsWith("blob:null/")) {
+                    // Fix uri when download from a HTML page file stored in local storage
+                    uri = uri.replaceFirst("^blob:null/", "blob:http://localhost/");
```

Comment:
Could you explain the reason for this change? How can we test it?

I had a look at the documentation and it seems that there is some discussion about how to implement `blob:null` and the origin of blobs in general:

- Blob spec: https://w3c.github.io/FileAPI/
- Gecko bug: [A Blob URL object's origin property should not be "null"](https://bugzilla.mozilla.org/show_bug.cgi?id=1270451)
- URL standard issue: [Origin of blob URLs doesn't match what implementations do](https://github.com/whatwg/url/issues/127)
- Chromium bug: [Not allowed to load local resource: blob:null](https://bugs.chromium.org/p/chromium/issues/detail?id=379206)
- Probably

Reply:
> Could you explain the reason for this change? How can we test it?

This is for fixing the blob download. When the HTML file is opened via `file://` instead of `http(s)://`, the blob uri will become `blob:null/`, which will lead to exceptions `java.lang.IllegalArgumentException: Can only download HTTP/HTTPS URIs` at https://github.com/Igalia/wolvic/blob/155495612f68b1b85ee81dd02c925ebb78475826/app/src/common/shared/com/igalia/wolvic/downloads/DownloadsManager.java#L215 You can test by storing the following example as an HTML page, open it as a file in Wolvic, and try to download the JSON file. https://codesandbox.io/s/download-json-file-with-js-p9t1z